frontend/src/components/Upgrade.tsx

```diff
@@ -1,20 +1,60 @@
 import { IUpgrade, UpgradeStatus } from "../../../shared/types";
+import { RGBTripleToCSS } from "../../../shared/util";
+import UpgradeImage from "./UpgradeImage";
 
 interface IUpgradeProps {
   up: IUpgrade;
   status: UpgradeStatus;
   onBuy: (up: IUpgrade) => void;
+  currencyAmount: number;
 }
 
-const Upgrade = ({ up, status, onBuy }: IUpgradeProps) => {
-  return (
-    <div
-      className={`upgrade status-${status}`}
-      onClick={() => {
+const bgUnavailableRGBT = { r: 179, g: 176, b: 176 };
+const bgAvailableRGBT = { r: 0, g: 255, b: 255 };
+const bgOwnedRGBT = { r: 100, g: 255, b: 0 };
+
+const bgAvailableCSS = RGBTripleToCSS(bgAvailableRGBT);
+const bgOwnedCSS = RGBTripleToCSS(bgOwnedRGBT);
+
+const Upgrade = ({ up, status, onBuy, currencyAmount }: IUpgradeProps) => {
```

Comment:
Maybe someone will know this for my curiosity: what's the mechanism that redraws the upgrade container? Does Phaser continually redraw components, or is it triggered by certain events (like selling data or buying upgrades)? I couldn't find any redraw calls in the event bus but maybe I'm missing something. I'm just wondering if there are any scenarios where this constructor isn't called and the background colors aren't updated to reflect the latest state.

Reply:
So for this part, we are actually not inside Phaser - it's React. So essentially it re-renders when one of the props changes (in this case, the `up`, `status`, `onBuy`, or `currencyAmount`). So as long as one of those things has changed, the upgrade appearance should be up-to-date